src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -160,20 +165,39 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             }
         }
 
-        // TODO would be more efficient to allow API to consolidate writeLog with exitStatus (save an RPC call)
-        @Override public Integer exitStatus(FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-            FilePath status = getResultFile(workspace);
-            if (status.exists()) {
-                try {
-                    return Integer.parseInt(status.readToString().trim());
-                } catch (NumberFormatException x) {
-                    throw new IOException("corrupted content in " + status + ": " + x, x);
+        /** Avoids excess round-tripping when reading status file. */
+        static class StatusCheck extends MasterToSlaveFileCallable<Integer> {
```

Comment:
This is just gratuitous merge conflict creation against #60. Better to keep the patch short and to the point.

Reply:
We have one particular user who has longer latencies and this change likely helps keep them under the timeouts - thus it is hardly "gratuitous."